packages/frontend/src/contexts/attestation.js

```diff
@@ -0,0 +1,56 @@
+import { makeAutoObservable } from 'mobx'
+import { SERVER } from '../config'
+
+export default class Attestation {
+  epoch = ''
+  epochKey = ''
+  index = ''
+  attester = ''
+  attesterId = ''
+  posRep = 0
+  negRep = 0
+  graffiti = ''
+  timestamp = ''
+  hash = ''
+  _id = ''
+
+  constructor(state) {
+    makeAutoObservable(this)
+    this.state = state
+    if (typeof window !== 'undefined') {
+      this.load()
+    }
+  }
+
+  async load() {
+    const url = new URL('api/attestation', SERVER)
```

Comment:
So here you should make a function like `loadAttestations` that does this logic. Then call that function in a component.

Then store the attestations in a variable like `attestationsByAttesterId`, or something like that. You should structure it however is most useful for the frontend. And you can have multiple kinds of structures, mabye like `allAttestations`, `attestationsById`, etc.

Generally though you want to avoid duplicating the objects. So if you have multiple stores you should use ids. So like `attestationsByAttesterId` stores a list of attestation ids, then to access it you have a function like this:

```js
attestationsByAttesterId(attesterId) {
  const attestationIds = this._attestationsByAttesterId
  return attestationIds.map(id => this.attestationsById(id))
}
```

This will reduce the amount of memory the webapp uses and prevent problems where objects aren't updated everywhere. But to start you can simply duplicate the objects